src/View.Uno/View.Uno.csproj

```diff
@@ -1,114 +1,110 @@
-		<Company>nventive</Company>
-		<AssemblyName>Nventive.View.Uno</AssemblyName>
-		<PackageId>Nventive.View.Uno</PackageId>
-		<Description>Nventive.View.Uno</Description>
-		<GenerateDocumentationFile>true</GenerateDocumentationFile>
-		<_Uno_XamlMerge_Task_Version>1.1.0-dev.13</_Uno_XamlMerge_Task_Version>
-		<XamlMergeOutputFile>$(MSBuildProjectDirectory)\Themes\MergedResourceDictionaries.xaml</XamlMergeOutputFile>
+	<!-- Change the TargetFrameworks depending on which platform you are building on. This avoids errors as it is impossible to build UAP on OSX (MacOS) -->
+	<TargetFrameworks Condition="'$([MSBuild]::IsOsPlatform(OSX))'">netstandard2.0;xamarinios10;</TargetFrameworks>
+	<TargetFrameworks Condition="'!$([MSBuild]::IsOsPlatform(OSX))'">netstandard2.0;xamarinios10;monoandroid11.0;monoandroid12.0;uap10.0.19041</TargetFrameworks>
+	<!-- Ensures the .xr.xml files are generated in a proper layout folder -->
+	<GenerateLibraryLayout>true</GenerateLibraryLayout>
+	<RootNamespace>Nventive.View</RootNamespace>
+	<Authors>nventive</Authors>
+	<Company>nventive</Company>
+	<AssemblyName>Nventive.View.Uno</AssemblyName>
+	<PackageId>Nventive.View.Uno</PackageId>
+	<Description>Nventive.View.Uno</Description>
+	<GenerateDocumentationFile>true</GenerateDocumentationFile>
+	<PackageTags>mvvm;xamarin;ios;android;uwp;chinook;</PackageTags>
+	<_Uno_XamlMerge_Task_Version>1.1.0-dev.13</_Uno_XamlMerge_Task_Version>
+	<XamlMergeOutputFile>$(MSBuildProjectDirectory)\Themes\MergedResourceDictionaries.xaml</XamlMergeOutputFile>
   </PropertyGroup>
 
   <PropertyGroup Condition="'$(TargetFramework)'=='netstandard2.0'">
     <DefineConstants>$(DefineConstants);__WASM__</DefineConstants>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Uno.SourceGenerationTasks" Version="4.0.0" />
+    <PackageReference Include="Uno.SourceGenerationTasks" Version="4.1.0" />
     <PackageReference Include="Uno.UI" Version="4.0.7" />
     <PackageReference Include="Uno.Core" Version="4.0.1" />
     <PackageReference Include="Uno.UI.Adapter.Microsoft.Extensions.Logging" Version="4.0.7" />
-	  <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="3.1.0" />
-	  <PackageReference Include="System.Reactive" Version="4.4.1" />
-	  <PackageReference Include="Uno.XamlMerge.Task">
-		  <Version>$(_Uno_XamlMerge_Task_Version)</Version>
-	  </PackageReference>
+	<PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="3.1.0" />
+	<PackageReference Include="System.Reactive" Version="4.4.1" />
+	<PackageReference Include="Uno.XamlMerge.Task">
+		<Version>$(_Uno_XamlMerge_Task_Version)</Version>
+	</PackageReference>
   </ItemGroup>
 
-	<ItemGroup>
-		<XamlMergeInput Include="Controls\**\*.xaml" />
-	</ItemGroup>
-
-	<ItemGroup>
-		<Reference Include="System.Xml" />
-		<Reference Include="System.Xml.Linq" />
-	</ItemGroup>
+  <ItemGroup>
+	<XamlMergeInput Include="..\..\Shared\Controls\**\*.xaml" />
```

Comment:
will this really work? I feel like each head should have its own file, but I might be mistaken

Reply:
I've tested it for uno and it does generate the control in the `MergedResourceDictionaries.xaml`

Reply:
based on the proposed changes of your PR, it does not generate the file properly

Reply:
The file is supposed to be in the .gitignore tho no?